extras/check_code_style.sh

```diff
@@ -0,0 +1,228 @@
+#!/bin/bash
+
+# static code analysis
+START_SHA=master
+END_SHA=HEAD
+
+CPPCHECK=cppcheck
+VERA=vera++
+
+CLANG_FORMAT=clang-format-3.6
+INCREMENTAL=false
+
+ONLY_FILE=
+
+#
+# usage [exit_code bad_option]
+#
+usage ()
+{
+    if test $1 -ne 0 ; then
+        echo "Unknown option: $2"
+    fi
+
+    cat <<EOF
+Usage: ./extras/check_code_style.sh [options ...]"
+
+This script checks our coding style guidelines. The checks are
+performed in the same way as in our TravisCI setup. First it looks
+for changed files in the repository in the commit range <git-start>..<git-end>
+(by default this is master..HEAD). Only the changed files are checked.
+You can specify a different commit range or define the file, which
```

Comment:
You can specify a different commit range, or define the file to be checked, as an argument